### PR TITLE
Updates, bugfix, static demo

### PR DESCRIPTION
**Pull request by wassname**
- updated: I updated/migrated to the latest versions of all packages. 
- new features:
  - sorting using CNTRL+UP and CNTRL+DOWN
  - folding when you click on a bullet point
  - works without a server: uses localstorage of webSQL when no server is present
- bugs: I fixed a few bugs, notably parent vs parentID. 
- bower: Switched to bower for frontend deps. 
- uses marionette to do some of the work of recursive rendering
- github page: I also made a fallback static demo so that it can be viewed as a github pages. [Example here](http://wassname.github.io/HackFlowy/)

Cheers,
wassname

### CODE REVIEW

**Comment by desyncr** [comment]
I'm take a look tomorrow. I looks good.

Thanks a lot. Amazing changes!

**Comment by captn3m0** [comment]
>github page: I also made a fallback static demo so that it can be viewed as a github pages.
This is epic. Does it still need a running heroku instance to fetch the data from?

**Comment by wassname** [comment]
Thanks! No need for heroku for the demo, it just uses some json demo data I put in the code. 

I was considering making it more like [laverna](https://laverna.cc/app/) which uses localstorage and syncs with dropbox which is a really nice setup. It would be pretty easy to add [backbone.localStorage](http://documentup.com/jeromegn/backbone.localStorage) to hackflowy to persist the lists in the user browser.

.. but I get to distracted from what I should be working on :)

**Comment by wassname** [comment]
This fixes Issue #20 and #2

**Comment by abhshkdz** [comment]
Thank you @wassname for contributing. :-) Yes, we should clone the `gh-pages` branch as well. @desyncr are you looking into this? If everything looks fine let's merge this soon.

**Comment by desyncr** [comment]
@abhshkdz I'm testing this out. Everything looks fine so far so I'll merge it today.

File: db/models/task.js [hunk 1]
```diff
@@ -5,10 +5,11 @@ module.exports = {
         task = orm.define('Tasks', {
           id         : {type: Sequelize.INTEGER, primaryKey: true, autoIncrement: true},
           content 	 : {type: Sequelize.TEXT, allowNull: false},
-          parent     : Sequelize.INTEGER,
-          isCompleted: Sequelize.BOOLEAN,
+          parentId   : {type: Sequelize.INTEGER, defaultValue: 0},
```
Comment (desyncr) [inline db/models/task.js]: Is this parentId or parent? as one on the post-install scripts  (db/seed/initial_tasks.js) complains about "table Task has no  column named  parentId".

File: db/seed/initial_tasks.js [hunk 2]
```diff
@@ -2,12 +2,12 @@ var config = require('config'),
...
     Tasks = require('../../db/models/task').instance(orm);
 
-Tasks.destroy().success(function() {
+Tasks.destroy({where: {}}).then(function() {
     Tasks.bulkCreate([
         {content: 'Welcome to HackFlowy!', isCompleted: false},
         {content: 'An open-source WorkFlowy clone', isCompleted: false},
         {content: 'Built using Backbone + Socket.IO', isCompleted: false},
-        {content: 'I pulled this together in a few hours to learn Backbone', isCompleted: false},
+        {content: 'Desyncr pulled this together in a few hours to learn Backbone', isCompleted: false},
```
Comment (desyncr) [inline db/seed/initial_tasks.js]: That wasn't me actually :P

**Comment by desyncr** [comment]
I see no problems for a merge. Unless somebody see something to note I'll merge at the end of the day.

File: db/seed/initial_tasks.js (hunk 2 above)
Comment (wassname) [inline db/seed/initial_tasks.js]: Ops sorry, I guess it was abhshkdz. I'll change that.

**Comment by wassname** [comment]
Thanks for that

**Comment by desyncr** [comment]
Wow such a mess. Well, it's reverted. The issue now is the repository history is a complete mess of commits. I'll rebase and push --force to restore everything to d3e38e461796563039e6a2297a551264f74a01db.

A new PR will be necessary. Thanks again @wassname and sorry for this inconvenience.

**Comment by desyncr** [comment]
Ok everything back to normal. @wassname you may now create another PR.

**Comment by wassname** [comment]
I'll do a new PR, sorry about that, thanks for fixing it!

**Pull request merged**